src/API/Base.php

```diff
@@ -116,7 +116,15 @@ public static function make_request( $endpoint, $method = 'POST', $payload = arr
 			return $response;
 		} catch ( Exception $e ) {
 
-			Logger::log( $e->getMessage(), 'error' );
+			if ( ! empty( Pinterest_For_WooCommerce()::get_setting( 'enable_debug_logging' ) ) ) {
+				/* Translators: 1: Error message 2: Stack trace */
+				Logger::log( sprintf( "%1\$s\n%2\$s", $e->getMessage(), $e->getTraceAsString() ), 'error' );
+			} else {
+				$pinterest_code = method_exists( $e, 'get_pinterest_code' ) ? '( pinterest code: ' . $e->get_pinterest_code() . ')' : '';
+
+				/* Translators: 1: Request method 2: Request endpoint 3: Response status code 4: Response message 5: Pinterest code */
+				Logger::log( sprintf( esc_html__( "%1\$s Request: %2\$s\nStatus Code: %3\$s\nAPI response: %4\$s%5\$s", 'pinterest-for-woocommerce' ), $method, $request['url'], $e->getCode(), $e->getMessage(), $pinterest_code ), 'error' );
```

Comment:
This line should be split across multiple lines for better readability.

Also, if we're going to look at the `$pinterest_code` variable as something relevant, why not have a separate `catch` statement for the `PinterestApiException` object where we know that method will exist?

Reply:
@JPry about the multi catch statement, would be a nice touch from a code perspective.
However, having 2 separate calls to `Logger`, will create 2 `ERROR` lines (with the time header), not sure if this will cause some kind of confusion, wdyt?

Reply:
@alexiglesias31 The two separate error logs would be within separate `catch` statements, so there wouldn't be duplicate log statements. Therefore I don't think there will be much confusion 🙂